compiler/rustc_session/src/config.rs

```diff
@@ -2678,7 +2678,12 @@ pub fn parse_crate_types_from_list(list_list: Vec<String>) -> Result<Vec<CrateTy
                 "cdylib" => CrateType::Cdylib,
                 "bin" => CrateType::Executable,
                 "proc-macro" => CrateType::ProcMacro,
-                _ => return Err(format!("unknown crate type: `{part}`")),
+                _ => {
+                    return Err(format!(
+                        "unknown crate type: `{part}`, expected one of: {display}",
+                        display = ["lib", "rlib", "staticlib", "dylib", "cdylib", "bin", "proc-macro"].join(", ")
```

Comment:
I suspect you may have to run `./x fmt` locally to format this line, but maybe not?